src/lib/Select.svelte

```diff
@@ -1,90 +1,316 @@
+        loadingIconSnippet,
+        itemSnippet,
+        selectionSnippet,
+        emptySnippet,
+        clearIconSnippet,
+        multiClearIconSnippet,
+        chevronIconSnippet,
+        inputHiddenSnippet,
+        requiredSnippet,
+        justValue = $bindable(),
+        useJustValue = false,
+        filter = _filter,
+        getItems = _getItems,
+        id = null,
+        name = null,
+        container = undefined,
+        input = undefined,
+        multiple = false,
+        multiFullItemClearable = false,
+        disabled = false,
+        focused = $bindable(false),
+        value = $bindable(),
+        filterText = $bindable(),
+        placeholder = 'Please select',
+        placeholderAlwaysShow = false,
+        items = null,
+        label = 'label',
+        itemFilter = (label, filterText, option) => `${label}`.toLowerCase().includes(filterText?.toLowerCase()),
+        groupBy = undefined,
+        groupFilter = (groups) => groups,
+        groupHeaderSelectable = false,
+        itemId = 'value',
+        loadOptions = undefined,
+        containerStyles = '',
+        listStyle = '',
+        hasError = false,
+        filterSelectedItems = true,
+        required = false,
+        closeListOnChange = true,
+        clearFilterTextOnBlur = true,
+        createGroupHeaderItem = (groupValue, item) => {
+            return {
+                value: groupValue,
+                [label]: groupValue,
+            };
+        },
+        getFilteredItems = () => {
+            return filteredItems;
+        },
+        searchable = true,
+        inputStyles = '',
+        clearable = true,
+        loading = $bindable(false),
+        listOpen = $bindable(false),
+        debounceWait = 300,
+        hideEmptyState = false,
+        inputAttributes = {},
+        listAutoWidth = true,
+        showChevron = false,
+        listOffset = 5,
+        hoverItemIndex = $bindable(0),
+        floatingConfig = {},
+        debounce = (fn, wait = 1) => {
+            clearTimeout(timeout);
+            timeout = setTimeout(fn, wait);
+        },
+        ariaValues = (values) => {
+            return `Option ${values}, selected.`;
+        },
+        ariaListOpen = (label, count) => {
+            return `You are currently focused on option ${label}. There are ${count} results available.`;
+        },
+        ariaFocused = () => {
+            return `Select is focused, type to refine list, press down to open the menu.`;
+        },
+        handleClear = () => {
+            clearState = true;
+            onclear(value);
+            value = undefined;
+            closeList();
+            handleFocus();
+        },
+        oninput = () => {},
+        onfilter = () => {},
+        onclear = () => {},
+        onfocus = () => {},
+        onchange = () => {},
+        onselect = () => {},
+        onblur = () => {},
+        onhoveritem = () => {},
+        onerror = () => {},
+        onloaded = () => {},
```

Comment:
Let these be `undefined`, and then in code use `?.()` to call them.  Example: `onfocus?.()`.